src/Service.Host/client/src/components/Pallets.js

```diff
@@ -0,0 +1,596 @@
+                                    resultLimit={100}
+                                    value={pallet?.defaultStockPoolDescription}
+                                    loading={stockPoolSearchLoading}
+                                    handleValueChange={(_, newVal) =>
+                                        setPallet({
+                                            ...pallet,
+                                            defaultStockPoolDescription: newVal
+                                        })
+                                    }
+                                    search={searchStockPool}
+                                    searchResults={stockPoolSearchResults}
+                                    priorityFunction="closestMatchesFirst"
+                                    onResultSelect={handleStockPoolSearchResultSelect}
+                                    clearSearch={clearStockPool}
+                                />
+                            </Grid>
+                        </Grid>
+                    </Grid>
+                    <Grid size={12}>
+                        <Grid container spacing={2}>
+                            <Grid size={2}>
+                                <Dropdown
+                                    value={pallet?.stockType}
+                                    fullWidth
+                                    propertyName="stockType"
+                                    label="Stock Type"
+                                    allowNoValue
+                                    items={stockTypes}
+                                    onChange={handleFieldChange}
+                                />
+                            </Grid>
+                            <Grid size={2}>
+                                <Dropdown
+                                    value={pallet?.stockState}
+                                    fullWidth
+                                    propertyName="stockState"
+                                    label="Stock State"
+                                    allowNoValue
+                                    items={stockStates}
+                                    onChange={handleFieldChange}
+                                />
+                            </Grid>
+                            <Grid size={2}>
+                                <Dropdown
+                                    value={pallet?.mixStates}
+                                    fullWidth
+                                    label="Mix States"
+                                    propertyName="mixStates"
+                                    allowNoValue
+                                    items={['Y', 'N']}
+                                    onChange={handleFieldChange}
+                                />
+                            </Grid>
+                        </Grid>
+                    </Grid>
+                    <Grid>
+                        <Grid size={12}>
```

Comment:
too many grids. you should only really need at most a nested grid depth of two (one top level container, then the items inside) unless you are trying to achieve really complicated subdivided layouts